#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: bash-4.0 upstream fix 026

			     BASH PATCH REPORT
			     =================

Bash-Release:	4.0
Patch-ID:	bash40-026

Bug-Reported-by:	Sergei Steshenko <sergstesh@yahoo.com>
Bug-Reference-ID:	<670181.38883.qm@web35204.mail.mud.yahoo.com>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-bash/2009-05/msg00059.html

Bug-Description:

A forgotten line in externs.h caused compilation errors to occur on some
systems (e.g., Cygwin).

Patch:

*** ../bash-4.0-patched/externs.h	2009-01-18 18:29:29.000000000 -0500
--- externs.h	2009-06-02 09:05:40.000000000 -0400
***************
*** 193,196 ****
--- 193,198 ----
  
  /* Declarations for functions defined in lib/sh/fpurge.c */
+ 
+ #if defined NEED_FPURGE_DECL
  #if !HAVE_DECL_FPURGE
  
***************
*** 201,205 ****
  
  #endif /* HAVE_DECL_FPURGE */
! 
  
  /* Declarations for functions defined in lib/sh/getcwd.c */
--- 203,207 ----
  
  #endif /* HAVE_DECL_FPURGE */
! #endif /* NEED_FPURGE_DECL */
  
  /* Declarations for functions defined in lib/sh/getcwd.c */
*** ../bash-4.0/patchlevel.h	2009-01-04 14:32:40.000000000 -0500
--- patchlevel.h	2009-02-22 16:11:31.000000000 -0500
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 25
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 26
  
  #endif /* _PATCHLEVEL_H_ */
